modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js

```diff
@@ -862,6 +855,19 @@ const FrontendDataSet = ({
 			});
 	}
 
+	const onSearch = ({query}) => {
+		if (apiURL || appURL) {
+			setSearchParam(query);
+		}
+		else {
+			setItems(
+				itemsProp.filter((item) => {
+					return JSON.stringify(item).includes(query);
+				})
+			);
+		}
+	};
```

Comment:
This is implementing search for the direct type of FDS usage.

Checking stringified item may have some edge cases with odd behavior, where item contains additional stuff than shown in UI, but I think simplicity of code makes it worth it. It can also result in unexpectedly "smart" search behavior :slightly_smiling_face:

Reply:
Yeah, on second thought this is not good, I'll limit check to just what is in UI.

Reply:
In the end, I limit the search to `Object.values(item)`, see https://github.com/liferay-frontend/liferay-portal/pull/4582/commits/651d6c302c976baa50d1ace397209ebeb2ebfb72. Having search fully customized to parts of `item` is not simple, as `item` is different for each visualization mode. I could hardcode it to list, but it would not work at all for table. This is a larger problem that we may want to tackle in a dedicated story, maybe add new API to define parts of item object to search. Even backend search of typical headless usage is not ideal, for example date does not match even though it is in a columns.